packages/react-components/src/components/chart/useTrendCursors.ts

```diff
@@ -32,7 +32,18 @@ const useTrendCursors = (
       chart?.getZr().on('click', (e) => {
         if (isInCursorAddMode && graphic.length < MAX_TREND_CURSORS) {
           setGraphic(
-            addNewTrendCursor(e, size, graphic.length, graphic, setGraphic, series, yMin, yMax, viewport, chart)
+            addNewTrendCursor(
+              e,
+              size,
+              trendCursorStaticIndex++,
+              graphic,
+              setGraphic,
+              series,
+              yMin,
+              yMax,
+              viewport,
+              chart
+            )
```

Comment:
Have you considered using an object-based API, i.e., `addTrendCursor({ e, size, [..] })`? It would remove the need to define arguments in order.

Reply:
yeah, if I add any more properties, il update this to an object, leaving it as is for now. No specific reason